web/html/src/manager/admin/config/monitoring.js

```diff
@@ -0,0 +1,250 @@
+// @flow
+import { hot } from 'react-hot-loader';
+import React from 'react';
+import {Panel} from 'components/panels/Panel';
+import {Button, AsyncButton} from 'components/buttons';
+import Network from 'utils/network';
+import {Messages, Utils as MessagesUtils} from 'components/messages';
+import {Utils} from 'utils/functions';
+import {IconTag as Icon} from 'components/icontag';
+
+import type JsonResult from "../../../utils/network";
+
+const {capitalize} = Utils;
+
+type Props = {
+
+}
+
+type ExportersStatus = {
+  exporters: {[string]: boolean}
+}
+
+type State = {
+  loading: boolean,
+  exportersStatus: {[string]: ?boolean},
+  monitoringEnabled: boolean,
+  messages: Array<Object>,
+  error: boolean,
+  loadingButtonText: string
+}
+
+const msgMap = {
+  "internal_error": t("An internal error has occured. See the server logs for details."),
+  "enabling_failed": t("Enabling monitoring failed. See the server logs for details."),
+  "enabling_failed_partially": t("Failed to enable all monitoring services. Some services are still disabled. See the server logs for details."),
+  "disabling_failed": t("Disabling monitoring failed. See the server logs for details."),
+  "disabling_failed_partially": t("Failed to disable all monitoring services. Some services are still enabled. See the server logs for details."),
+  "enabling_succeeded": t("Monitoring enabled successfully."),
+  "disabling_succeeded": t("Monitoring disabled successfully."),
+  "no_change": t("Monitoring status hasn't changed."),
+  "unknown_status": t("Monitoring status unknown.")
+};
+
+const exporterMap = {
+  "node": t("System"), 
+  "tomcat": t("Tomcat (Java JMX)"),
+  "taskomatic": t("Taskomatic (Java JMX)"),
+  "postgres": t("PostgreSQL database")
+}
+
+function nullify(exporters: {[string]: ?boolean}) : {[string]: ?boolean} {
```

Comment:
it's just to avoid hard-coding exporter keys. Currently only the names of the exporters and the initial state are hard-coded. The backend could return additional exporters and the only change required in the UI would be add some user friendly names and a key to the initial state.

Reply:
Removed function.